res/device-mqtt/configuration.toml

```
@@ -1,96 +1,126 @@
-BootTimeout = 30000
-Protocol = 'http'
-CheckInterval = '10s'
-ConnectRetries = 100
-ClientMonitor = 15000
-Labels = []
-StartupMsg = 'device simple started'
-Timeout = 5000
-EnableAsyncReadings = true
-AsyncBufferSize = 16
+HealthCheckInterval = "10s"
+Host = "localhost"
+Port = 59982
+ServerBindAddr = ""  # blank value defaults to Service.Host value
+StartupMsg = "device mqtt started"
+# MaxRequestSize limit the request body size in byte of put command
+MaxRequestSize = 0 # value 0 unlimit the request size.
+RequestTimeout = "5s"
 
 [Registry]
-Host = 'edgex-core-consul'
+Host = "localhost"
 Port = 8500
-Type = 'consul'
+Type = "consul"
 
 [Clients]
-  [Clients.Data]
-  Name = 'edgex-core-data'
-  Protocol = 'http'
-  Host = 'edgex-core-data'
-  Port = 48080
-  Timeout = 50000
+  [Clients.core-data]
+  Protocol = "http"
+  Host = "localhost"
+  Port = 59880
+
+  [Clients.core-metadata]
+  Protocol = "http"
+  Host = "localhost"
+  Port = 59881
+
+[MessageQueue]
+Protocol = "mqtt"
+Host = "localhost"
+Port = 6379
+Type = "mqtt"
+AuthMode = "none"
+PublishTopicPrefix = "edgex/events/device" # /<device-profile-name>/<device-name>/<source-name> will be added to this Publish Topic prefix
+  [MessageQueue.Optional]
+  ClientId = "device-mqtt"
+  Qos = "0" # Quality of Sevice values are 0 (At most once), 1 (At least once) or 2 (Exactly once)
+  KeepAlive = "10" # Seconds (must be 2 or greater)
+  Retained = "false"
+  AutoReconnect = "true"
+  ConnectTimeout = "5" # Seconds
+  SkipCertVerify = "false" # Only used if Cert/Key file or Cert/Key PEMblock are specified
 
-  [Clients.Metadata]
-  Name = 'edgex-core-metadata'
-  Protocol = 'http'
-  Host = 'edgex-core-metadata'
-  Port = 48081
-  Timeout = 50000
+[SecretStore]
+Type = "vault"
+Host = "localhost"
+Port = 8200
+Path = "device-mqtt/"
+Protocol = "http"
+RootCaCertPath = ""
+ServerName = ""
+# Uncomment to seed service secrets file
+# SecretsFile = "/tmp/edgex/secrets/device-mqtt/secrets.json"
+DisableScrubSecretsFile = false
+TokenFile = "/tmp/edgex/secrets/device-mqtt/secrets-token.json"
+  [SecretStore.Authentication]
+  AuthType = "X-Vault-Token"
+  [SecretStore.RuntimeTokenProvider]
+  Enabled = false
+  Protocol = "https"
+  Host = "localhost"
+  Port = 59841
+  TrustDomain = "edgexfoundry.org"
+  EndpointSocket = "/tmp/edgex/secrets/spiffe/public/api.sock"
+  RequiredSecrets = "redisdb"
 
 [Device]
-  DataTransform = true
-  InitCmd = ''
-  InitCmdArgs = ''
-  MaxCmdOps = 128
-  MaxCmdValueLen = 256
-  RemoveCmd = ''
-  RemoveCmdArgs = ''
-  ProfilesDir = '/res'
-  UpdateLastConnected = false
+DataTransform = true
+MaxCmdOps = 128
+MaxCmdValueLen = 256
+ProfilesDir = "./res/profiles"
+DevicesDir = "./res/devices"
+UpdateLastConnected = false
+Labels = []
+EnableAsyncReadings = true
+AsyncBufferSize = 16
+UseMessageBus = true
 
-[Logging]
-EnableRemote = false
-File = ''
+[MQTTBrokerInfo]
+Schema = "tcp"
+Host = "localhost"
+Port = 1883
+Qos = 0
+KeepAlive = 3600
+ClientId = "device-mqtt"
```

Has to be different name than the connection for the messagebus above
```suggestion
ClientId = "device-mqtt-external"
```